app/models/conversion_host.rb

```diff
@@ -12,14 +10,16 @@ class ConversionHost < ApplicationRecord
 
   validates :name, :presence => true
   validates :resource_id, :presence => true
-  validates :resource_type, :presence => true, :inclusion => { :in => VALID_RESOURCE_TYPES }
+  validates :resource_type, :presence => true
 
   validates :address,
     :uniqueness => true,
     :format     => { :with => Resolv::AddressRegex },
     :inclusion  => { :in => ->(conversion_host) { conversion_host.resource.ipaddresses } },
     :unless     => ->(conversion_host) { conversion_host.resource.blank? || conversion_host.resource.ipaddresses.blank? }
 
+  validate :resource_supports_conversion_host
```

Comment:
It would be really cool if we could `delegate :supports_conversion_host?, :to => :resource` then just `validate supports_conversion_host?` but it looks like supports_feature_mixin can't do this type of validation yet....

Reply:
Reasonable idea. Let's investigate this in a separate, refactoring PR.

Reply:
I noticed that Rails 5.1 has `delegate_missing_to`, so perhaps we can point that to `resource` once we're on 5.1+.